### PR TITLE
feat(ui): show configured Git ref in UI

### LINKED ISSUE #2672: [FLI-780] Show configured Git ref in UI
### Problem

When running with our Git backend [https://www.flipt.io/docs/configuration/storage#git](https://www.flipt.io/docs/configuration/storage#git), it would be nice to show what the current 'tracked' branch / tag / ref is that Flipt is configured with, alongside the `Read Only` pill. See 👇🏻 for mockup

### Ideal Solution

CleanShot 2024-01-17 at 10 41 12

* show the configured ref in the UI, (in the screenshot above I'm using DB storage so the icon is wrong, but this should only be enabled if running with Git backend)
* on hover of the branch/ref, maybe we display the full path (URL) to the repo and branch/ref?

### Search

- [X] I searched for other open and closed issues before opening this

### Additional Context

This info, the current ref, should already be served by the `/meta/config` endpoint by the API. This is how we already determine which storage icon to display, and if to put the UI in readonly mode.

[https://github.com/flipt-io/flipt/blob/0fd3fc1c53c210897610f2e0559c64a3c01d3933/ui/src/components/header/ReadOnly.tsx#L22](https://github.com/flipt-io/flipt/blob/0fd3fc1c53c210897610f2e0559c64a3c01d3933/ui/src/components/header/ReadOnly.tsx#L22)

<sub>From [SyncLinear.com](https://synclinear.com) | [FLI-780](https://linear.app/flipt/issue/FLI-780/show-configured-git-ref-in-ui)</sub>

### PR DESCRIPTION
**Pull request by erka**
- long `ref` will be truncated with **...** if too long (`max-width: 8rem; /* 128px */`)
- the icons were changed so git will be more understandable.

fixes #2672

### CODE REVIEW

File: ui/src/components/header/ReadOnly.tsx [hunk 1]
```diff
@@ -32,8 +32,20 @@ export default function ReadOnly() {
         config.storage?.type || 'unknown'
       )} Storage`}
     >
-      {StorageIcon && (
-        <StorageIcon className="h-4 w-4 fill-gray-200" aria-hidden="true" />
+      {config.storage?.type == 'git' && (
+        <span
+          className="max-w-32 truncate font-medium"
```
Comment (markphelps) [inline ui/src/components/header/ReadOnly.tsx:37]:
```suggestion
          className="nightwind-prevent text-violet-200 max-w-32 truncate font-medium"
```

wdyt?

CleanShot 2024-01-17 at 16 47 03@2x

**Review by markphelps — APPROVED** [review]
looks amazing @erka !! just what we had in mind.

one q, what do you think about giving the ref a slightly different shade? I suggested one in the PR

File: ui/src/components/header/ReadOnly.tsx (hunk 1 above)
Comment (erka) [reply ui/src/components/header/ReadOnly.tsx]: it looks great!

**Review by markphelps — APPROVED** [review]
amazing!!

**Pull request merged**